python/pyspark/ml/param/__init__.py

```diff
@@ -444,7 +444,14 @@ def _setDefault(self, **kwargs):
         Sets default params.
         """
         for param, value in kwargs.items():
-            self._defaultParamMap[getattr(self, param)] = value
+            p = getattr(self, param)
```

Comment:
In a previous PR a parameter was given an incorrect type converter, and this was not caught by the tests. Enforcing `_setDefault` to use the type converter for the param will ensure that all params with default values cannot be given incompatible type converters.